src/Components/FitnessPage.js

```diff
@@ -0,0 +1,280 @@
+import React, { useState, useEffect, useRef, useLayoutEffect } from 'react';
+import { Platform, View, StyleSheet, Text, TouchableOpacity, FlatList, ScrollView, Share } from 'react-native';
+
+import { MaterialCommunityIcons, Ionicons } from '@expo/vector-icons';
+
+import I18n from '../utils/i18n';
+
+import Animation from '../Components/animation/Animation';
+import VimeoVideo from '../Components/shared/VimeoVideo';
+import theme from '../styles/theme.style';
+
+import Drill from '..//Components/animation/Drill';
+import { createLink } from '../utils/firebase';
+
+const FitnessPage = (props) => {
+  const { route, navigation } = props;
+
+  const drill = props.route.params.drill;
+
+  const share = async (drill) => {
+    const url = await createLink(
+      'drills/' + drill.id,
+      drill.title,
+      I18n.t('drillPage.description', { description: drill.description.slice(0, 70) }),
+    );
+
+    const youtubeVideos = drill.steps.reduce((total, step) => {
+      const stepvideo = step.youtube ? `${step.title} - ${step.youtube}\n` : '';
+      return total + stepvideo;
+    }, '');
+
+    Share.share({
+      title: I18n.t('drillPage.shareTitle', { drillTitle: drill.title }),
+      message: I18n.t('drillPage.shareContent', { url, youtubeVideos, count: youtubeVideos.length }),
+      url,
+    }).catch((e) => console.log(e));
+  };
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      title: drill.title,
+      headerTintColor: theme.COLOR_PRIMARY_LIGHT,
+      headerStyle: { backgroundColor: theme.COLOR_PRIMARY },
+      headerRight: () => (
+        <TouchableOpacity onPress={() => share(drill)} testID="shareButton">
+          <Ionicons
+            name={Platform.select({
+              ios: 'ios-share-outline',
```

Comment:
If we use it several times we should extract it to another component, maybe in the `/drills` subfolder